backend/src/jobs_server/utils/kueue.py

```diff
@@ -177,17 +177,54 @@ def pod(self) -> client.V1Pod:
 
         if self.owner_uid is None:
             self.owner_uid = self.managed_resource.metadata["uid"]
-        owner_uid = self.owner_uid
-        podlist: client.V1PodList = api.list_pod_for_all_namespaces(
-            label_selector=f"controller-uid={owner_uid}"
-        )
-        pods = podlist.items
+
+        if self.managed_resource.kind == "Job":
+            # Jobs are simple, we can use the `controller-uid` label to find the associated pods.
+
+            owner_uid = self.owner_uid
+            podlist: client.V1PodList = api.list_pod_for_all_namespaces(
+                label_selector=f"controller-uid={owner_uid}"
+            )
+            pods = podlist.items
+        elif self.managed_resource.kind == "RayJob":
+            # RayJobs have an additional layer of indirection:
+            #
+            # The Kuberay operator creates a RayCluster resource for the job,
+            # and the pods (head, worker) are in turn created for that RayCluster.
+            #
+            # Then, a submission job is created to submit the Ray job to the RayCluster.
+            # The job is identified by two labels:
+            #
+            # - `ray.io/originated-from-crd=RayJob`
+            # - `ray.io/originated-from-cr-name=<ray-job-name>`
+            #
+            # Once we know the job, we can find the pods as usual.
+
+            rayjob_name = self.metadata.owner_references[0].name
+            submission_jobs: client.BatchV1JobList = client.BatchV1Api().list_job_for_all_namespaces(
+                label_selector=f"ray.io/originated-from-crd=RayJob,ray.io/originated-from-cr-name={rayjob_name}",
+            )
+            submission_jobs = submission_jobs.items
+
+            if not submission_jobs:
+                return None
+            if len(submission_jobs) > 1:
+                raise RuntimeError(
+                    f"more than one submission job found for RayJob {rayjob_name!r}: {submission_jobs!r}"
+                )
+
+            podlist: client.V1PodList = api.list_pod_for_all_namespaces(
```

Comment:
So the members in `submission_jobs` are actually manifests of the k8s jobs themselves? This is not exactly clear before from the variable names.